shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/NearbyStaticData.kt

```diff
@@ -466,6 +533,214 @@ fun NearbyStaticData.withRealtimeInfo(
         .sortedWith(compareBy(Route.relevanceComparator(pinnedRoutes)) { it.sortRoute() })
 }
 
+/**
+ * For reasons, for mid-line subway shuttles/suspensions, schedules will correctly reflect the
+ * closure and use route patterns that stop at temporary terminals, but predictions will use the
+ * canonical route patterns. The way we work around this is that for
+ * 1. subway routes
+ * 2. with alerts somewhere on the line
+ * 3. where the schedule has non-typical patterns instead of typical patterns
+ * 4. but the predictions all have typical patterns instead of non-typical patterns
+ *
+ * we check the stops of the scheduled patterns against both the predicted pattern and the
+ * prediction's current stop to override the prediction's route pattern with a correct scheduled
+ * pattern, and then we load the headsign from the new route pattern's representative trip. This
+ * should be narrowly scoped enough to apply to temporary terminals but no other trips. Hopefully.
+ */
+private fun NearbyStaticData.rewrittenForTemporaryTerminals(
+    predictions: PredictionsStreamDataResponse?,
+    globalData: GlobalResponse?,
+    alerts: List<Alert>?,
+    schedules: ScheduleResponse?,
+): Pair<NearbyStaticData, PredictionsStreamDataResponse?> {
```

Comment:
suggestion: This is some pretty complicated logic - I think it would be worth breaking this out into named & documented helper functions.

Reply:
Is the path I'm on now of having nested helper functions inside `NearbyStaticData.rewrittenForTemporaryTerminals` reasonable, or would you rather have non-nested helper functions (perhaps even with their own unit tests)? The main advantage of nested helper functions is they can refer to the `globalData` without needing to take it as a parameter, but that may not be worth it.

Reply:
I think non-nested helpers would be easier to read, and unit tests for meaningfully complex helpers would be nice too.

Reply:
I have moved all the helpers into a separate `TemporaryTerminalRewriter` class which also encapsulates all the state, and I've written unit tests for all the individual pieces.